kf_lib_data_ingest/validation/values.py

```diff
@@ -0,0 +1,236 @@
+    },
+    CONCEPT.SEQUENCING.PAIRED_END: {"1", "2"},
+    CONCEPT.GENOMIC_FILE.SIZE: check_non_negative,
+    # INVESTIGATOR
+    # OUTCOME
+    CONCEPT.OUTCOME.EVENT_AGE_DAYS: check_age,
+    CONCEPT.OUTCOME.DISEASE_RELATED: {
+        constants.OUTCOME.DISEASE_RELATED.YES,
+        constants.OUTCOME.DISEASE_RELATED.NO,
+    },
+    CONCEPT.OUTCOME.VITAL_STATUS: {
+        constants.OUTCOME.VITAL_STATUS.ALIVE,
+        constants.OUTCOME.VITAL_STATUS.DEAD,
+    },
+    # PHENOTYPE
+    CONCEPT.PHENOTYPE.EVENT_AGE_DAYS: check_age,
+    CONCEPT.PHENOTYPE.OBSERVED: {
+        constants.PHENOTYPE.OBSERVED.YES,
+        constants.PHENOTYPE.OBSERVED.NO,
+    },
+    # READ_GROUP
+    CONCEPT.READ_GROUP.LANE_NUMBER: check_positive,
+    CONCEPT.READ_GROUP.QUALITY_SCALE: {
+        constants.READ_GROUP.QUALITY_SCALE.ILLUMINA13,
+        constants.READ_GROUP.QUALITY_SCALE.ILLUMINA15,
+        constants.READ_GROUP.QUALITY_SCALE.ILLUMINA18,
+        constants.READ_GROUP.QUALITY_SCALE.SANGER,
+        constants.READ_GROUP.QUALITY_SCALE.SOLEXA,
+    },
+    # READ_GROUP_GENOMIC_FILE
+    # SEQUENCING_CENTER
+    # SEQUENCING_EXPERIMENT
+    CONCEPT.SEQUENCING.DATE: check_date,
+    CONCEPT.SEQUENCING.STRATEGY: {
+        constants.SEQUENCING.STRATEGY.TARGETED,
+        constants.SEQUENCING.STRATEGY.WXS,
+        constants.SEQUENCING.STRATEGY.WGS,
+        constants.SEQUENCING.STRATEGY.RNA,
+        constants.SEQUENCING.STRATEGY.MRNA,
+        constants.SEQUENCING.STRATEGY.LINKED_WGS,
+        constants.COMMON.OTHER,
+    },
+    # CONCEPT.SEQUENCING.LIBRARY_PREP - there is no concept for this
+    # CONCEPT.SEQUENCING.LIBRARY_SELECTION - no concept for this either
+    CONCEPT.SEQUENCING.LIBRARY_STRAND: {
+        constants.SEQUENCING.STRAND.UNSTRANDED,
+        constants.SEQUENCING.STRAND.FIRST,
+        constants.SEQUENCING.STRAND.SECOND,
+        constants.COMMON.OTHER,
+    },
+    CONCEPT.SEQUENCING.MAX_INSERT_SIZE: check_positive,
+    CONCEPT.SEQUENCING.MEAN_DEPTH: check_positive,
+    CONCEPT.SEQUENCING.MEAN_INSERT_SIZE: check_positive,
+    CONCEPT.SEQUENCING.MEAN_READ_LENGTH: check_positive,
+    CONCEPT.SEQUENCING.PLATFORM: {
+        constants.SEQUENCING.PLATFORM.ION_TORRENT,
+        constants.SEQUENCING.PLATFORM.SOLID,
+        constants.SEQUENCING.PLATFORM.ILLUMINA,
+        constants.SEQUENCING.PLATFORM.LS454,
+        constants.SEQUENCING.PLATFORM.GENOMICS,
+        constants.SEQUENCING.PLATFORM.PACBIO,
+        constants.COMMON.OTHER,
+    },
+    CONCEPT.SEQUENCING.TOTAL_READS: check_positive,
+    # SEQUENCING_EXPERIMENT_GENOMIC_FILE
+    # STUDY
+    # STUDY_FILE
+    CONCEPT.STUDY_FILE.AVAILABILITY: {
+        constants.GENOMIC_FILE.AVAILABILITY.IMMEDIATE,
+        constants.GENOMIC_FILE.AVAILABILITY.COLD_STORAGE,
+    },
+    # TASK
+    # TASK_GENOMIC_FILE
+}
+
+
+INPUT_VALIDATION = {
+    k: (
+        (
+            f"Must be one of {(v | NULL_VALUES)}",
+            lambda x: x in (v | NULL_VALUES),
```

Comment:
Oh no. this isn't the same kind of function as the wrapped ones.

uhhhh....hmm.

What if we make it so that the innermost part of try_wrap does `ret = func(x)`, then change the wrapped functions to return bool instead of assert, and then wrap this lambda with `try_wrap(message)(thelambda)` instead of this tuple